client/consensus/src/lib.rs

```diff
@@ -131,19 +136,39 @@ impl<B, I, C> BlockImport<B> for FrontierBlockImport<B, I, C> where
 		if self.enabled {
 			let log = find_frontier_log::<B>(&block.header)?;
 			let hash = block.post_hash();
+			let number: u64 = UniqueSaturatedInto::<u64>::unique_saturated_into(
+				*block.header.number()
+			);
 
 			match log {
 				ConsensusLog::EndBlock {
 					block_hash, transaction_hashes,
 				} => {
 					aux_schema::write_block_hash(client.as_ref(), block_hash, hash, insert_closure!());
-
-					for (index, transaction_hash) in transaction_hashes.into_iter().enumerate() {
-						aux_schema::write_transaction_metadata(
-							transaction_hash,
-							(block_hash, index as u32),
-							insert_closure!(),
-						);
+					if transaction_hashes.len() > 0 {
+						if let Some(pending) = &self.pending_transactions {
```

Comment:
I suggest we move this to a block import notification, as this pending transaction removal logic does not really belong to consensus. Can be done in a separate PR though.

Reply:
Sure sounds good, it was one of the other options to do it. Although, isn't having a dedicated service background task just for that a bit of a overkill?

Reply:
I moved the queue maintenance to the service https://github.com/paritytech/frontier/pull/243/files#diff-584822e937c9489fbcbee1642b3b01f204fe6304b7c0ff091f55d750bff3f615R233.